src/core/surface/channel_create.c

```diff
@@ -200,7 +201,7 @@ grpc_channel *grpc_insecure_channel_create(const char *target,
       "grpc_insecure_channel_create(target=%p, args=%p, reserved=%p)", 3,
       (target, args, reserved));
   GPR_ASSERT(!reserved);
-  if (grpc_channel_args_is_census_enabled(args)) {
+  if (grpc_channel_args_is_census_enabled(args) || census_enabled()) {
```

Comment:
Shouldn't this be && ? Or can the census_enabled() check be dropped, and made part of whatever grpc_channel_args_is_census_enabled checks?